sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBAccount.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.Provisioning.ResourceManager;
+using Azure.ResourceManager.CosmosDB;
+using Azure.ResourceManager.CosmosDB.Models;
+namespace Azure.Provisioning.CosmosDB
+{
+    /// <summary>
+    /// Represents a RedisCache.
+    /// </summary>
+    public class CosmosDBAccount : Resource<CosmosDBAccountData>
+    {
+        private const string ResourceTypeName = "Microsoft.DocumentDB/databaseAccounts";
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="CosmosDBAccount"/> class.
+        /// </summary>
+        /// <param name="scope"></param>
+        /// <param name="kind"></param>
+        /// <param name="consistencyPolicy"></param>
+        /// <param name="accountOfferType"></param>
+        /// <param name="accountLocations"></param>
+        /// <param name="parent"></param>
+        /// <param name="name"></param>
+        /// <param name="version"></param>
+        /// <param name="location"></param>
+        public CosmosDBAccount(
+            IConstruct scope,
+            CosmosDBAccountKind? kind = default,
```

Comment:
In general if these items are settable on the Properties I think we should not have them in the constructor as optional as it makes the ctor very large in some cases.

Reply:
kind seems pretty fundamental though, but I will keep that in mind.